src/effects/deeplearning/ModelManagerPanel.cpp

```diff
@@ -0,0 +1,809 @@
+
+#include <wx/scrolwin.h>
+#include <wx/range.h>
+#include <wx/gauge.h>
+#include <wx/button.h>
+#include <wx/sizer.h>
+#include <wx/textdlg.h>
+#include <wx/stattext.h>
+#include <wx/hyperlink.h>
+
+
+#include "AllThemeResources.h"
+#include "Theme.h"
+
+#define MODELCARDPANEL_WIDTH 600
+#define MODELCARDPANEL_HEIGHT 150
+#define DETAILEDMODELCARDPANEL_WIDTH 400
+#define DETAILEDMODELCARDPANEL_HEIGHT 400
+#define MODELCARDPANEL_X_OFFSET 20
+#define MANAGERPANEL_WIDTH (MODELCARDPANEL_WIDTH + DETAILEDMODELCARDPANEL_WIDTH + MODELCARDPANEL_X_OFFSET)
+
+// ModelManagerPanel
+// TODO: need to get rid of the unique ptrs to UI elements
+ModelManagerPanel::ModelManagerPanel(wxWindow *parent, EffectDeepLearning *effect)
+                                    : wxPanelWrapper(parent)
+{
+   mEffect = effect;
+   mTools = nullptr;
+
+   ShuttleGui S(this, eIsCreating);
+   PopulateOrExchange(S);
+   Layout();
+   Fit();
+   Center();
+}
+
+void ModelManagerPanel::PopulateOrExchange(ShuttleGui & S)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   S.StartVerticalLay(true);
+   {
+      mTools = safenew ManagerToolsPanel(S.GetParent(), this);
+      // mTools->PopulateOrExchange(S);
+      S.AddWindow(mTools);
+
+      S.StartMultiColumn(2, wxEXPAND);
+      {
+         mScroller = S.StartScroller(wxVSCROLL | wxTAB_TRAVERSAL);
+         {
+         }
+         S.EndScroller();
+         wxSize size(MODELCARDPANEL_WIDTH+50, DETAILEDMODELCARDPANEL_HEIGHT);
+         wxSize vsize(MODELCARDPANEL_WIDTH+MODELCARDPANEL_X_OFFSET, 
+                     DETAILEDMODELCARDPANEL_HEIGHT);
+         mScroller->SetVirtualSize(vsize);
+         mScroller->SetSize(size); 
+         mScroller->SetMinSize(size); 
+         mScroller->SetMaxSize(size);
+         mScroller->SetWindowStyle(wxBORDER_SIMPLE);
+         mScroller->SetScrollRate(0, 10);
+
+         // S.SetStretchyCol(1);
+         mDetailedPanel = safenew DetailedModelCardPanel(
+               S.GetParent(), wxID_ANY, manager.GetEmptyCard(), mEffect, this);
+
+         S.AddWindow(mDetailedPanel);
+
+      }
+   }
+   S.EndVerticalLay();
+
+   FetchCards();
+}
+
+// TODO
+void ModelManagerPanel::Clear()
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   for (auto const& pair : mPanels)
+   {
+      ModelCardHolder card = pair.second->GetCard();
+      if (manager.IsInstalling(card))
+         manager.CancelInstall(card);
+   } 
+
+   // clean up panels
+   mPanels.clear();
+}
+
+void ModelManagerPanel::AddCard(ModelCardHolder card)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+
+   mScroller->EnableScrolling(true, true);
+   std::string repoId = card->GetRepoID();
+   mPanels[repoId] = std::make_unique<SimpleModelCardPanel>(mScroller, wxID_ANY,
+                                                             card, mEffect, this);
+
+   // if this is the first card we're adding, go ahead and select it
+   if (mPanels.size() == 1)
+   {
+      mEffect->SetModel(card);
```

Comment:
std::move?

Reply:
why use std::move if `card` is a shared_ptr?